runtime/centrifuge/src/migrations.rs

```diff
@@ -142,36 +145,34 @@ mod burn_unburned {
 		}
 
 		fn on_runtime_upgrade() -> Weight {
-			let data = orml_tokens::Accounts::<T>::get(
+			let free_balance = orml_tokens::Accounts::<T>::mutate_exists(
 				<Domain as Convert<_, T::AccountId>>::convert(ETH_DOMAIN),
 				LP_ETH_USDC,
+				|maybe_data| {
+					let free_balance = maybe_data
+						.clone()
+						.map(|account_data| account_data.free)
+						.unwrap_or_default();
+					*maybe_data = None;
+					free_balance
+				},
 			);
+			orml_tokens::TotalIssuance::<T>::remove(LP_ETH_USDC);
```

Comment:
If we remove the total issuance, should not iterate through all accounts (`Accounts::iter()`)  ensuring that there are no other accounts than `ETH_DOMAIN` containing `LP_ETH_USDC`?

Reply:
In the `pre_upgrade` hook of `try-runtime`, we ensure that the total issuance matches the balance of the domain account. But you are right that this protect the `on_runtime_upgrade` hook. I think by moving the `burn_unburned` migration after the asset registry migrations, it works with the previous code in which we only burn from the domain account!

Reply:
I guess the issue only happens if just another account than `ETH_DOMAIN` mints some `LP_ETH_USDC` before the upgrade. If we know this can not occur until before applying the RU, then we can go as it is right now 👍🏻. The easier for you!

Reply:
By reordering, I was able to use the old code: [ed84b56](https://github.com/centrifuge/centrifuge-chain/pull/1747/commits/ed84b562a17e9ccf907a6a3ab53b0f1848285dfc)